cmd/kola/kola.go

```diff
@@ -310,7 +314,23 @@ func runList(cmd *cobra.Command, args []string) {
 		fmt.Fprintln(w, "Test Name\tPlatforms\tArchitectures\tDistributions")
 		fmt.Fprintln(w, "\t")
 		for _, item := range testlist {
-			fmt.Fprintf(w, "%v\n", item)
+			platformFound := false
+			for _, platform := range item.Platforms {
+				if listPlatform == "all" || platform == "all" || platform == listPlatform {
+					platformFound = true
+					break
```

Comment:
Hmm, I think we can check `listPlatform` upfront to see if it's `all`, right? That way we avoid iterating entirely.

Whenever I encounter nested for-loops like this and bool vars, I always check if it'd be cleaner to use functions. So e.g. here it'd be:

```go
for _, item := range testlist {
  if matchesFilter(item.Platforms, listPlatforms) && matchesFilter(item.Distros, listDistro) {
    fmt.Fprintf(...)
  }
}
```

That way, the functions can just `return true` when found and plus you dedupe the loop logic (and plus we don't try to filter by distro if the platform didn't match anyway).

Reply:
That's fair, if only golang had a native slice.Contains method 😄

Reply:
Updated with pre-checking `listPlatform` and `listDistro` :)

Reply:
> That's fair, if only golang had a native slice.Contains method smile

And of course...generics and... :wink: